addons/map/functions/fnc_blueForceTrackingUpdate.sqf

```diff
@@ -26,6 +26,28 @@ if (GVAR(BFT_Enabled) and {(!isNil "ACE_player") and {alive ACE_player}}) then {
         };
     };
 
+    if (GVAR(BFT_ShowPlayerNames)) then {
+        _playersToDrawMarkers = allPlayers select {side _x == _playerSide};
+
+        {
+            private _markerType = [_x] call EFUNC(common,getMarkerType);
+            private _colour = format ["Color%1", side _x];
+
+            private _marker = createMarkerLocal [format ["ACE_BFT_%1", _forEachIndex], [(getPos leader _x) select 0, (getPos leader _x) select 1]];
+            _marker setMarkerTypeLocal _markerType;
+            _marker setMarkerColorLocal _colour;
+            _marker setMarkerTextLocal (name _x);
+
+            GVAR(BFT_markers) pushBack _marker;
+        } forEach _playersToDrawMarkers;
+
+        _groupsToDrawMarkers = _groupsToDrawMarkers select {
```

Comment:
I'll wrap it in a condition that makes its purpose a bit clearer, what it does is filter players out of the _groupsToDrawMarkers array if ShowPlayerNames is true, so if HideAiGroups is false, AI groups draw normally while players draw as intended.